Prototype guidance draft pages

timsb opened the pull request.

Jon-Rowe-HMRC reviewed (changes requested):
http://localhost:3000/prototype-guidance/
'Installing the govuk prototype kit' > 'Installing the GOV.UK Prototype Kit'

Always refer to the prototype kit exactly as 'GOV.UK Prototype Kit'? There's a few variations across the pages.

http://localhost:3000/prototype-guidance/installing-prototype-kit/
'You should use the official guidance for getting started with the prototype kit' > 'Use the official guidance for getting started with the prototype kit'?

http://localhost:3000/prototype-guidance/vs-code/
"VS Code is a free, open-source code editor available for Windows, macOS, and Linux.
Follow these steps to set up Visual Studio Code (VS Code) for designing with the GOV.UK Prototype Kit." the order of shortening VS Code is incorrect, and inconsistent elsewhere on the page. Just always 'Visual Studio Code'?

'File > Open' and 'Settings' should have curled quote marks

http://localhost:3000/prototype-guidance/getting-started-html-css/
Slightly scary how basic we're going on some of this page, but hey-ho :)

http://localhost:3000/prototype-guidance/using-github/
Replace the hyphen with an em dash
Github are inconsistent on their guidance title capitalisation, but we're more inconsistent ('glossary'?)
 
http://localhost:3000/prototype-guidance/version-control/
We're not consistently following the GDS bullet style of not using full-stops or defaulting to starting with lowercase
 
 http://localhost:3000/prototype-guidance/show-different-pages-based-on-answers/
 'GOV.UK design system' > 'GOV.UK Design System' x3
 'Under' > 'under' x2
 'In this case' > 'in this case'
 
 http://localhost:3000/prototype-guidance/where-to-go-for-support/
 'senior interaction designer' > 'Senior Interaction Designer'

timsb commented:
> http://localhost:3000/prototype-guidance/ 'Installing the govuk prototype kit' > 'Installing the GOV.UK Prototype Kit'
>
> Always refer to the prototype kit exactly as 'GOV.UK Prototype Kit'? There's a few variations across the pages.

When talking about the 'Prototype kit' I've renamed them all to 'GOV.UK Prototype Kit' for consistency however when just referring to 'prototype(s)' I've kept that as is. @JackHMRC could you check that I've changed the correct ones?

> http://localhost:3000/prototype-guidance/installing-prototype-kit/ 'You should use the official guidance for getting started with the prototype kit' > 'Use the official guidance for getting started with the prototype kit'?

I've changed this.

> http://localhost:3000/prototype-guidance/vs-code/ "VS Code is a free, open-source code editor available for Windows, macOS, and Linux. Follow these steps to set up Visual Studio Code (VS Code) for designing with the GOV.UK Prototype Kit." the order of shortening VS Code is incorrect, and inconsistent elsewhere on the page. Just always 'Visual Studio Code'?

The official documentation is inconsistent in how it shortens it (or not). I suggest we start with 'Visual Studio Code (VS Code)' to start with and then carry on with 'VS Code' as it's useful to know the shortened version as most people refer to it as that. @JackHMRC any thoughts?
> 'File > Open' and 'Settings' should have curled quote marks

Done, found a few more on that page as well which I've corrected.

> http://localhost:3000/prototype-guidance/getting-started-html-css/ Slightly scary how basic we're going on some of this page, but hey-ho :)

🥲

> http://localhost:3000/prototype-guidance/using-github/ Replace the hyphen with an em dash Github are inconsistent on their guidance title capitalisation, but we're more inconsistent ('glossary'?)

I think you were just suggesting lowercase 'g' for 'glossary'? If so, I've done this.

> http://localhost:3000/prototype-guidance/version-control/ We're not consistently following the GDS bullet style of not using full-stops or defaulting to starting with lowercase

@JackHMRC I think we discussed this with Beth at the time. Can you see if we're correct on this page?

> http://localhost:3000/prototype-guidance/show-different-pages-based-on-answers/ 'GOV.UK design system' > 'GOV.UK Design System' x3 'Under' > 'under' x2 'In this case' > 'in this case'

Done

> http://localhost:3000/prototype-guidance/where-to-go-for-support/ 'senior interaction designer' > 'Senior Interaction Designer'

Done

JackHMRC commented:
Responses to Jon's points: 

1. "Simplify to this maybe?
'Shortcut to use it:

Shift + Alt + F (Windows)
Shift + Option + F (macOS)'"

- Happy with this suggestion

2. "http://localhost:3000/prototype-guidance/getting-started-html-css/
'Start learning some of the basics of HTML and CSS with free access' is missing a full-stop"

- Agree

3. "http://localhost:3000/prototype-guidance/version-control/
'automated routing and redirects..." etc

- Agree to start lowercase after each dash

4. "routes.js - this is the central routing file, it mounts each version’s router under its respective path (such as, /v1 or /v2).
routing.js - these files define the routes and ensure that all navigation stays within the correct version."

- I'd swap the colons for em dashes in these two bullets given that we've used them above

5. "Same issue under 'Automatic redirects'."

- I'd adopt Jon's suggestion just to have this in paragraphed form. Suggested text:

"This ensures that redirects always include the correct version prefix. If you call res.redirect('/question-2'), it automatically updates to /v1/question-2 or /v2/question-2, depending on the version. This code should go in the routing.js file for each version.

You must type out the full path for the question page and the redirect page (for example, /question-1 or /question-2). You do not need to include v1 or v2—this is handled automatically. However, if your version folder contains subdirectories, you should include the full nested path (such as, /nested/question-1). The subRouter function should be used instead of the main router function.

6. "http://localhost:3000/prototype-guidance/show-different-pages-based-on-answers/
Full stops should be added to a few sentences on this page?"

Two paragraphs under the heading "Routing radio buttons" need a full stop. The second para under "Routing checkboxes" needs one too.

timsb commented:
Thanks for the confirmation @JackHMRC.

I've changed all those things apart from the bit under 'Automatic redirects' The three bullet points are almost like added points or steps to be aware of or even instructions. I'm not sure adding the first bullet point to the first paragraph ('This ensures that redirects...') and then putting the last two bullet points into their own paragraph quite works.

The bullet points are distinct from the first paragraph as the the first paragraph is an explanation of what the code does and the bullet points are how to achieve / where to put that code.

I think the first paragraph ('This ensures that redirects...') needs to stand on its own but then the bullet points could either stay as bullet points or become their own sentence. I'm kind of leaning towards the former to keep as bullet points as they're distinct steps to achieve the goal but happy to go with whatever you think.

Jon-Rowe-HMRC reviewed (approved):
Looks good, approved :)

One tiny thing...
http://localhost:3000/prototype-guidance/using-github/
“branch”, “commit”, and “merge” are more suited to being in quote, not speech marks?
